src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -455,7 +456,7 @@ private void EnsureTransactionsAreSupported()
             {
                 if (connectedDataBearingServer.Type == ServerType.ShardRouter)
                 {
-                    Feature.ShardedTransactions.ThrowIfNotSupported(connectedDataBearingServer.Version);
+                    throw new NotSupportedException($"Server version {connectedDataBearingServer.Version} does not support the ShardedTransactions feature.");
```

Comment:
This doesn't look quite right.  If the server is a 4.2 server, it does support transactions.  The problem is that the _driver_ does not support transactions on a sharded cluster.

Reply:
Good catch. See new commit.